src/functions/module_math.ts

```diff
@@ -996,6 +997,56 @@ export const RAND: AddFunctionDescription = {
   isExported: true,
 };
 
+// -----------------------------------------------------------------------------
+// RANDARRAY
+// -----------------------------------------------------------------------------
+export const RANDARRAY: AddFunctionDescription = {
+  description: _lt("Returns a grid of random numbers between 0 inclusive and 1 exclusive."),
+  args: [
+    arg("rows (number, default=1)", _lt("The number of rows to be returned.")),
+    arg("columns (number, default=1)", _lt("The number of columns to be returned.")),
+    arg("min (number, default=0)", _lt("The minimum number you would like returned.")),
+    arg("max (number, default=1)", _lt("The maximum number you would like returned.")),
+    arg("whole_number (number, default=FALSE)", _lt("Return a whole number or a decimal value.")),
+  ],
+  returns: ["RANGE<NUMBER>"],
+  compute: function (
+    rows: PrimitiveArgValue = 1,
+    columns: PrimitiveArgValue = 1,
+    min: PrimitiveArgValue = 0,
+    max: PrimitiveArgValue = 1,
+    whole_number: PrimitiveArgValue = false
+  ): number[][] {
+    const _cols = toNumber(columns);
+    const _rows = toNumber(rows);
+    const _min = toNumber(min);
+    const _max = toNumber(max);
+    const _whole_number = toBoolean(whole_number);
+
+    assertPositive(_lt("The number columns (%s) must be positive.", _cols.toString()), _cols);
+    assertPositive(_lt("The number rows (%s) must be positive.", _rows.toString()), _rows);
+    assert(
+      () => _min <= _max,
+      _lt(
+        "The maximum (%s) must be greater than or equal to the minimum (%s).",
+        _max.toString(),
+        _min.toString()
+      )
+    );
+
+    const result: number[][] = Array(_cols);
+    for (let col = 0; col < _cols; col++) {
+      result[col] = Array(_rows);
+      for (let row = 0; row < _rows; row++) {
+        const random = Math.random() * (_max - _min) + _min;
+        result[col][row] = _whole_number ? Math.round(random) : random;
```

Comment:
the distribution is no longer uniform when applying the round directly in this way :)